### ISSUE TITLE
Asset System Registration

### ISSUE THREAD

**Issue by calclavia** [NOVA-Team/NOVA-Core#157]
Maybe I should consider making the asset system do auto-registration for all assets? I think it's quite redundant to be registering locations of textures and sounds when clearly, it's not going to change or do anything.

Reference to assets should be just done in a ResourceLocation fashion (we will call it Asset(domain,name)

For example:
Block Textures: BlockTexture extends Asset
Sound: Sound extends Asset

BlockTexture lives in assets/domain/textures/blocks as defined by the BlockTexture path, but essentially ANY texture pack should be usable if you define it yourself.

**Comment by Caellian** [comment]
It again might be one of those "Minecraft-only" things. Most of the games
store resources directly in resources folder (root of the jar), Minecraft
added assets and domain folders to make resources more organised.

You should probably make path a non-final variable as it might change for
different games in which case it should be changed by game wrapper.

**Comment by calclavia** [comment]
Yes, but since NOVA is a mod system, domains are supported.

**Comment by Caellian** [comment]
This won't cause issues with accessing default game resources for other
games?

**Comment by calclavia** [comment]
I don't think so. You access the default game by using the domain "minecraft" or whatever game it's called.
